src/components/ActivitiesList/ActivitiesDataFetch.tsx

```diff
@@ -0,0 +1,52 @@
+import { Activity } from "@components/ActivitiesList/ActivitiesList";
+import axios from "axios";
+import React, { ReactElement, useEffect, useState } from "react";
+
+import styles from "./ActivitiesList.module.css";
+
+const URL = "http://localhost:8000/api/v1/activities";
+
+const ActivitiesList: React.FC = (): ReactElement => {
+  const [activities, setActivities] = useState<Activity[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const {
+          data: { activities },
+        } = await axios.get(URL);
+        setActivities(activities);
+      } catch (error) {
+        console.error("Error fetching activities:", error);
+        if (axios.isAxiosError(error)) {
+          return {
+            data: "API request failed! Did you remember to `npm run dev` the backend app?",
+          };
+        }
+        return {
+          data: "A fetching data error occurred.",
+        };
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  return (
+    <div className={styles.activitiesContainer}>
+      <h1>Activities</h1>
+      <div className={styles.activitiesList}>
+        {activities.map((activity) => (
+          <div key={activity._id} className={styles.activityItem}>
+            <p>Activity Name: {activity.name}</p>
+            <p>Description: {activity.description}</p>
+            <p>Category: {activity.category}</p>
+            <p>Vote Count: {activity.votes}</p>
```

Comment:
This also could have changed since you submitted your PR, but I'm seeing that sometimes `votes` is entirely `undefined`. Because of that we'll get an error if we try to access `undefined.count`, so we can use the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and n[ullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) to make sure the activity with no `votes` object displays a zero instead of crashing the page:

```suggestion
              <p>Vote Count: {activity.votes?.count ?? 0}</p>
```

`votes?.count` is a shortcut for what used to have to be done in a ternary before this feature: 

`votes !== undefined ? votes.count : undefined`

then once we have either a number or the `undefined`, the `??` (a more modern version of `||` fallback) allows the value to fall back to `0`

Reply:
The shape of the vote has changed @dargaCode based on the changes requested for the backend schema. Should I hold the PR until the backend team finalizes it first?

Reply:
This is an example of a resource received from the Activity data model:
```
{
          "_id": "656b62e0fa36aea592c2308e",
          "name": "Movie Night",
          "description": "Gather friends and watch a movie.",
          "category": "social",
          "createdAt": "2023-12-02T17:01:20.844Z",
          "updatedAt": "2023-12-02T17:01:20.844Z",
          "vote": {
              "_id": "656b62e0fa36aea592c2308f",
              "activityID": "656b62e0fa36aea592c2308e",
              "voters": [
                  {
                      "_id": "656b62e6fa36aea592c23093",
                      "name": "Mo",
                      "email": "mo@email.com",
                      "createdAt": "2023-12-02T17:01:26.417Z",
                      "updatedAt": "2023-12-02T17:01:26.417Z",
                      "__v": 0
                  },
                  {
                      "_id": "6568fb622d91faf5b3b777e1",
                      "name": "Morez",
                      "email": "example@email.com",
                      "createdAt": "2023-11-30T21:15:14.363Z",
                      "updatedAt": "2023-11-30T21:15:14.363Z",
                      "__v": 0
                  }
              ],
              "createdAt": "2023-12-02T17:01:20.844Z",
              "updatedAt": "2023-12-02T17:02:02.943Z",
              "__v": 2
          },
          "__v": 0
},
```

Reply:
Thanks for resolving this you two!